pkg/sql/testdata/order_by

```diff
@@ -667,3 +667,23 @@ EXPLAIN (METADATA) UPDATE t SET c = TRUE RETURNING b
 2  scan                      (a, b, c)          +a,unique
 2          table  t@primary
 2          spans  ALL
+
+statement ok
+CREATE TABLE xyzw (
+  x INT PRIMARY KEY,
+  y INT,
+  z INT,
+  w INT,
+  INDEX ywxz (y, w, x, z),
+  INDEX ywz (y, w, z)
+)
+
+# Verify that the outer ordering is propagated to index selection and we choose
+# the index that avoids any sorting.
+query ITTTTT
+EXPLAIN (METADATA) SELECT * FROM (SELECT y, w, x FROM xyzw WHERE y = 1 ORDER BY w) ORDER BY w, x
+----
+0  render                    (y, w, x)              =y,+w,+x
```

Comment:
This would be the result without the change (we prefer `ywz` because it has fewer columns):
```
0  sort                     (y, w, x)              =y,+w,+x
0          order  +w,+x
1  render                   (y, w, x)              =y,+w
2  scan                     (x, y, z[omitted], w)  =y,+w,+z,+x,unique
2          table  xyzw@ywz
2          spans  /1-/2
```